File: Irrgarten/src/irrgarten/Weapon.java
package irrgarten;

/**
 *
 * @author juanma
 */
public class Weapon {

    private float power;
    
    private int uses;
    
    public Weapon (float iPower, int iUses){
        power = iPower;
        uses = iUses;
    }
    
    float attack(){
        if(uses > 0){
            uses--;
            return power;
        }
        return 0;
    }
    
    public String toString(){
        String _power = Float.toString(power);
        String _uses = Integer.toString(uses);
        String cadena  = "W[" + _power + ", " + _uses + "]";
        return cadena;
    }
    
}

File: IrrgartenRuby/Weapon.rb
module Irrgarten
  def class Weapon::
    def initialize(iPower, iUses)
      @power = iPower
      @uses = iUses
    end
    
    def attack
      if(@uses > 0) then
        @uses -= 1
        return @power
      end
      0
    end

    def to_s
      "W[#{@power}, #{@uses}]"
    end
    
    def discard
      Dice.discardElement(@uses)
    end


  end
  
end
